### PR TITLE
Vault operations on closed/empty vault

### PR DESCRIPTION
**Pull request by pacxiu**
image

## Changes 👷‍♀️
- add `hasToDepositCollateralOnEmptyVault` error + message
- add stories
- add tests
- add calculating initial stage for otherAction/adjustPosition if vault has 0 locked collateral.
  
## How to test 🧪
- test in UI/Storybook
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: public/locales/en/common.json [hunk 1]
```diff
@@ -253,7 +253,8 @@
       "depositing-all-eth-balance": "You need some ETH to pay for the gas fees. Lower your deposit amount",
       "ledger-enable-contract-data": "Please enable Contract data on the Ledgers Ethereum app Settings",
       "withdraw-collateral-on-vault-under-debt-floor": "You cannot withdraw from your vault while it is below the minimum required of {{debtFloor}} Dai. To withdraw your collateral, payback your debt fully. Learn more about the Minimum Vault Debt <0>here</0>.",
-      "exchange-error": "Could not fetch exchange prices for doing this multiply action."
+      "exchange-error": "Could not fetch exchange prices for doing this multiply action.",
+      "has-to-deposit-collateral-on-empty-vault": "There is no collateral in this vault. Please first add collateral in the 'Other actions' tab, with the 'deposit collateral' action"
```
Comment (johnnieskywalker) [inline public/locales/en/common.json:257]: You can add new key to other locales but we can do it later with script as well

**Review by johnnieskywalker — DISMISSED** [review]

File: features/manageMultiplyVault/manageMultiplyVault.ts [hunk 2]
```diff
@@ -397,17 +398,23 @@ function addTransitions(
...
-  selectedCollateralAllowanceRadio: 'unlimited',
-  selectedDaiAllowanceRadio: 'unlimited',
-  showSliderController: true,
-  closeVaultTo: 'collateral',
-  mainAction: 'buy',
-  otherAction: 'depositCollateral',
+export function defaultMutableManageMultiplyVaultState(
+  lockedCollateral?: BigNumber,
+): MutableManageMultiplyVaultState {
+  const hasZeroCollateral = lockedCollateral?.eq(zero)
```
Comment (adamskrodzki) [inline features/manageMultiplyVault/manageMultiplyVault.ts:404]: what if it is close to zero ? If some pennies are still in a vault ? does UI behave correctly then (I guess any collateralisation  different than infinity is possible due to dust limit)

Comment (pacxiu) [reply features/manageMultiplyVault/manageMultiplyVault.ts:404]: Yes, if it is close to zero you can perform all actions on the vault eg. payback all debt, validations about dust limit will apply then if you try to do something which will result in vault being in dust range after those actions are performed.

**Comment by johnnieskywalker** [comment]
I guess this behaviour is acceptable ass user can't perform any action until slider is moved and then everything is fine 
image

**Comment by johnnieskywalker** [comment]
image

:heavy_check_mark:

**Comment by johnnieskywalker** [comment]
image

:heavy_check_mark:

**Review by johnnieskywalker — APPROVED** [review]
tested and approved you can see screenshots in pull request also seems to solve problem mentioned by @paszkowskiDamian

**Comment by johnnieskywalker** [comment]
image
image

image

image


image
image

after clicking close this vault on bottom of form
image

**Review by adamskrodzki — APPROVED** [review]

**Pull request merged**